Implement GMT themes (sets of GMT defaults) (#3344)
GenericMappingTools/gmt: doc/rst/source/gmt.conf.rst

        Sets the upper limit on the number of cores any multi-threaded module might
        use (whether **-x** is selected or not) [0, i.e., as many as are available].

    **GMT_THEME**
        Override GMT default settings with those of the selected theme.  Choose from
        *classic* [Default for classic mode], *modern* [Default for modern mode],
        *inverse* (white on black), *movie* (suitable for movie making) and *darkmode*.
        You can also create and use your own themes by compiling files of desired settings

>         and place them in your GMT user directory (usually ~/.gmt) and name them *theme*.conf.

Reviewer: Better to place them in a subdirectory.
```suggestion
        and place them in your GMT user directory (usually ~/.gmt/themes) and name them *theme*.conf.
```

Author: Agreed, done.

        Sets the upper limit on the number of cores any multi-threaded module might
        use (whether **-x** is selected or not) [0, i.e., as many as are available].

    **GMT_THEME**
        Override GMT default settings with those of the selected theme.  Choose from
        *classic* [Default for classic mode], *modern* [Default for modern mode],

>         *inverse* (white on black), *movie* (suitable for movie making) and *darkmode*.
>         You can also create and use your own themes by compiling files of desired settings
>         and place them in your GMT user themes directory (usually ~/.gmt/themes) and name
>         them *theme*.conf.

Reviewer 2: ```suggestion
        and *minimal*. You can also create and use your own themes by compiling 
        files of desired settings and place them in your GMT user themes directory 
        (usually ~/.gmt/themes) and name them *theme*.conf.
```

        **e**, **s**, **n**, **z** (uppercase allowed as well). [we] (if nothing specified).
        Note that this setting can be overridden via the **+a** modifier in **-B**.

    **MAP_AUTO_SCALE**
        Determines if basemap font sizes, pen thicknesses, tick-lengths, and offsets will

>         be computed from the plot dimensions (**on**) or if they should be set manually (**off**) [off].

Author: done
